fixieai/agents/standalone.py

```diff
@@ -31,7 +33,7 @@ def __init__(
 
         self._handle_message = handle_message
         self._sample_queries = sample_queries
-        utils.validate_registered_pyfunc(handle_message, self)
+        utils.validate_registered_pyfunc(handle_message, self, is_generator=True)
```

Comment:
For my own understanding, is it a _requirement_ that the function be a generator/return an iterator? My only concern would be if this will trip folks up about needing to `yield` instead of `return` and so I'm wondering if there's a sensible way to tell if the function has returned a generator. (And/or infer it as such from the type signature)

Reply:
It's not a requirement, but we'd need to handle both cases differently. (Which admittedly should be fairly straightforward from the typing).

Reply:
Done.